### ISSUE TITLE
Importing any file in Linux 

### ISSUE THREAD

**Issue by Djaenk** [wwmm/easyeffects#1179]
### EasyEffects Version

6.1.2

### What package are you using?

Flatpak (Flathub)

### Distribution

Linux Mint 20.2 Cinnamon

### Expected Behavior

When importing parametric EQ presets or convolver impulses, a file explorer should open to let me browse to and select a file.

### Actual Behavior

Clicking the button to import presets or import impulses opens the file explorer at the root directory and freezes the entire desktop. I have to ctrl-alt-backspace to restart the X server to regain control.

**Comment by Djaenk** [comment]
I've identified that any button which creates a dialog box (Presets, settings menu, Add Plugin, etc.) freezes everything that isn't easyeffects. The rest of my system doesn't become responsive until I close the dialog by clicking somewhere else on easyeffects.

**Comment by vchernin** [comment]
See https://github.com/wwmm/easyeffects/issues/1165

I wonder what it is on Cinnamon/Mint that causes this.

**Comment by DianaNites** [comment]
Getting this in KDE Plasma, Arch Linux, sometimes.

**Comment by vchernin** [comment]
> Getting this in KDE Plasma, Arch Linux, sometimes.

I wonder why sometimes? Is there a way to reproduce the issue consistently?

**Comment by vchernin** [comment]
I was able to reproduce at least one issue with the Flatpak build (gtk 4.4). I can dismiss the popover by clicking anywhere on Wayland, but I need to click within the app on X11.

I think it is this issue: https://gitlab.gnome.org/GNOME/gtk/-/issues/3502

**Comment by wwmm** [comment]
Ok. They already know that their X11 backend has problems.

**Comment by abdumu** [comment]
> I've identified that any button which creates a dialog box (Presets, settings menu, Add Plugin, etc.) freezes everything that isn't easyeffects. The rest of my system doesn't become responsive until I close the dialog by clicking somewhere else on easyeffects.
> 
> My workaround is now to maximize easyeffects before trying to import anything so the file browser doesn't completely cover up the application. I can then close that dialog box and continue as normal.

I can confirm, there is a bug, and your workaround actually works. thanks.

ElementaryOS 6.

**Comment by RW21** [comment]
Can confirm that it occurs on popos gnome 3.38.4. 
>My workaround is now to maximize easyeffects before trying to import anything so the file browser doesn't completely cover up the application. I can then close that dialog box and continue as normal.

This workaround didn't work for me.

**Comment by vchernin** [comment]
Yes switching to Wayland avoids this.

Ubuntu 22.04 (releasing in a couple weeks) will default to Wayland so the impact of this will be a lot less widespread.

**Comment by wwmm** [comment]
>Any work arounds to importing the wav file without using the UI so I can try eq?

Yes. All that the import dialogs do is copying files to folders inside `~/.config/easyeffects/`. You can do that manually if necessary.

**Comment by BrianLeishman** [comment]
Whatever way I installed easyeffects placed them into this folder `~/.var/app/com.github.wwmm.easyeffects/config/easyeffects/irs`

**Comment by wwmm** [comment]
>Whatever way I installed easyeffects placed them into this folder ~/.var/app/com.github.wwmm.easyeffects/config/easyeffects/irs

This folder is used by the Flatpak package.

**Comment by ejuarezg** [comment]
I just came across this issue in PopOS 22.04. Just FYI, if you are using a FIR WAV file, you have to change the file extension from `.wav.` to `.irs` when moving into the `irs` folder listed in the previous comment.

**Comment by Peppershaker** [comment]
I'm still suffering from this issue on Ubuntu 22.04 LTS

**Comment by wwmm** [comment]
>I'm still suffering from this issue on Ubuntu 22.04 LTS

Yes. I have no idea if gtk develoers are looking at this. Since the issue https://gitlab.gnome.org/GNOME/gtk/-/issues/3502 was opened a few gtk releases have come. I haven't used X11 in a while. At least on Wayland the dialogs are working. But popover menus still have problems in some corner cases.

**Comment by zer0exia** [comment]
On EndeavourOS ( Kernel: 6.1.6-arch1-1) , this issue also occurs on Wayland KDE Plasma, in addition to X11. I guess by reading through that linked gitlab issue and related MR posts, looks like it's entirely possible that this will be a permanent issue for KDE.

**Comment by wwmm** [comment]
>looks like it's entirely possible that this will be a permanent issue for KDE.

I wonder why this is happening on KDE even using wayland. At least here on GNOME + wayland file dialogs are working. In any case at least for wayland gtk has to fix this. Unless there is something wrong on KDE that needs to be fixed on their side.

**Comment by zer0exia** [comment]
I take that back. Needs feedback from another KDE wayland users as well, to make sure whether this issue is only on my (wayland ) end or is actually a problem on all. Though this is on a fresh installed system, so I might have missed something. I hadn't touched easyeffects presets for a long while on previous system( also plasma wayland), so I had no idea the freezing on file-picker was an issue.

**Comment by wwmm** [comment]
As `gtk 4.10` deprecated `GtkFileChooserNative` in favor of `GtkFileDialog` I am starting to do the replacement in EasyEffects. The preset import dialog code in our master branch is now based on GtkFileDialog. I wonder if with it xorg users still have problems.

**Comment by wwmm** [comment]
Now that we moved to Qt debugging gtk issues do not make sense. Let's close this issue.
